go/vt/vtgate/executor.go

```diff
@@ -974,21 +976,21 @@ type iQueryOption interface {
 
 // getPlan computes the plan for the given query. If one is in
 // the cache, it reuses it.
-func (e *Executor) getPlan(ctx context.Context, vcursor *vcursorImpl, sql string, comments sqlparser.MarginComments, bindVars map[string]*querypb.BindVariable, qo iQueryOption, logStats *logstats.LogStats) (*engine.Plan, error) {
+func (e *Executor) getPlan(ctx context.Context, vcursor *vcursorImpl, sql string, comments sqlparser.MarginComments, bindVars map[string]*querypb.BindVariable, qo iQueryOption, logStats *logstats.LogStats) (*engine.Plan, sqlparser.Statement, error) {
```

Comment:
I'm thinking - WDYT about adding the AST to the `engine.Plan` struct instead of passing it as a second argument here? It kind of makes sense to have it available in the plan, right?

Reply:
We cache the plan, is caching the ast useful or not? If not, then we are just increasing the cache size or reducing the number of plans that can be cached.